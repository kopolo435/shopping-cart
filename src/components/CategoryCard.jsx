import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import React from "react";

function CategoryCard({ data }) {
  return (
    <Link to={data.link}>
      <div className="cardImg">
        <img src={data.img} alt="" />
      </div>
      <div className="cardContent">
        <h3>{data.name}</h3>
        <p>{data.description}</p>
      </div>
    </Link>
  );
}

CategoryCard.propTypes = {
  data: PropTypes.shape({
    name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    link: PropTypes.string.isRequired,
    img: PropTypes.string.isRequired,
  }).isRequired,
};
export default CategoryCard;
