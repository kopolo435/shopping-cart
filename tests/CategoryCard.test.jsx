import { render, screen } from "@testing-library/react";
import { expect, it } from "vitest";
import { MemoryRouter } from "react-router-dom";
import React from "react";
import CategoryCard from "../src/components/CategoryCard";

it("Renders correct Link card", () => {
  const testData = {
    name: "Cakes",
    description: "Los mejores pasteles",
    link: "/cakes",
    img: "#",
  };
  const { container } = render(
    <MemoryRouter>
      <CategoryCard data={testData} />
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});
