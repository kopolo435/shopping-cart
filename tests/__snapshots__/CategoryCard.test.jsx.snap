// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Renders correct Link card 1`] = `
<div>
  <a
    href="/cakes"
  >
    <div
      class="cardImg"
    >
      <img
        alt=""
        src="#"
      />
    </div>
    <div
      class="cardContent"
    >
      <h3>
        Cakes
      </h3>
      <p>
        Los mejores pasteles
      </p>
    </div>
  </a>
</div>
`;
